WordPress/gutenberg
# Term sorting is incorrect in flat term selector

## What?
The current search for term selection component is to search by most popular rather than a name search. With a limit of 20 terms to show up, this could mean the tag you are looking for does not show up on the list. This does not scale for sites with lots of terms of a taxonomy. We discovered it on one of our sites that has 13K tags.

## Why?
For sites with lots of tags, especially similarly spelled tags, it could result in the term not appearing in the autocomplete.

## How?
Since this is a search by name, it should have an orderby `name` and order `asc` to ensure the term you are searching for appears in the list.

## Testing Instructions
1. Add a bunch of tags with similar name (test, test1, test2, etc)
2. Assign one of those tags to a post (say `test3`)
3. When searching for tag and you type `test`, the first item will be `test3` with current code
4. In pull request the search will go by name / asc so it will be `test` as the first item (which is what someone would expect)

Current
<img width="294" alt="Screen Shot 2023-02-06 at 11 40 36 AM" src="https://user-images.githubusercontent.com/818980/217037183-5bd74155-214c-46d4-a704-c74c1b04000d.png">

Fix
<img width="294" alt="Screen Shot 2023-02-06 at 11 41 52 AM" src="https://user-images.githubusercontent.com/818980/217037230-68812f24-a179-4f8a-a257-0e3c2e666f8d.png">
